draft-ietf-quic-multipath.md

```diff
@@ -1424,26 +1406,39 @@ MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
 ~~~
 MAX_PATHS Frame {
   Type (i) = 0x15228c0b,
-  Maximum Paths (i),
+  Maximum Path Identifier (i),
 }
 ~~~
 {: #fig-max-paths-frame-format title="MAX_PATHS Frame Format"}
 
 MAX_PATHS frames contain the following field:
 
 Maximum Path Identifier:
-: A count of the cumulative number of path that can be opened
-  over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not
-  possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits
-  opening of a path with Path ID larger than this limit MUST be treated
-  as a connection error of type FRAME_ENCODING_ERROR.
+: The latest Path ID that can be used for this connection.
+  This value cannot exceed 2^32-1, as it is not
+  possible to encode Path IDs larger than 2^32-1.
+
+Receipt of PATH_AVAILABLE, PATH_STANDBY, PATH_ABANDON or MP_ACK frames
+that uses a Path ID that is larger than the announced Path ID
+MUST be treated as a connection error of type FRAME_ENCODING_ERROR.
+
 
 Loss or reordering can cause an endpoint to receive a MAX_PATHS frame with
-a lower path limit than was previously received. MAX_PATHS frames that
-do not increase the path limit MUST be ignored.
+a smaller Path ID than was previously received. MAX_PATHS frames that
+do not announce a larger Path ID than previously received MUST be ignored.
+
+Endpoints SHOULD NOT issue new connection IDs which have path identifiers larger than
+the Path ID announced in the Maximum Path Identifier field in the
+MP_MAX_PATHS frame {{max-paths-frame}}.
```

Comment:
I think we should clarify that the MAX_PATHS frame is a unidirectional advertisement to the peer having the same semantic as the `initial_max_paths` TP. To that end, an endpoint MUST NOT issue new CIDs for a Path ID larger than the maximum one advertised by the peer (and not SHOULD NOT) and MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION. Allowing (without recommending) an endpoint to issue CIDs with Path ID larger than supported by the receiving host pose security concerns from a resource exhaustion viewpoint.

However, if the peer allows a maximum Path ID X that is larger than the local advertised maximum path ID Y (i.e., X > Y), then an endpoint may (MAY?) provides CIDs for Path IDs Z where Y < Z <= X.

Further, we probably mention the MAX_PATHS frame (and not MP_MAX_PATHS).

Reply:
But yes, we really need to clarify that the value is unidirectional. I was thinking at some point that this is a global value and thought that if the MAX_PATHS frame gets reordered, a MP_NEW_CONNECTION_ID frame using a new path id could arrive before. However, as MAX_PATHS is sent from one peer and MP_NEW_CONNECTION_ID from the other, this cannot happen. So this is should be a MUST NOT. I will propose further text to clarify tomorrow!

```suggestion
Endpoints MUST NOT issue new connection IDs which have path identifiers larger than
the Path ID announced by the peer in the Maximum Path Identifier field in the
MP_MAX_PATHS frame {{max-paths-frame}}.
```